### PR TITLE
[INLONG-3885][Sort] Add KafkaExtractNode supporting

### LINKED ISSUE #3885: [Feature][Sort] Add KafkaExtracNode to support Kafka source
### Description

Add KafkaExtracNode for support source is kafka

### Use case

It will support extract data of kafka.

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by gong**
### [INLONG-3885][Sort]Add KafkaExtractNode supporting

Fixes #3885 

### Motivation

Add KafkaExtractNode supporting

### Modifications

Add KafkaExtractNode class and test class 
refactor test package

### CODE REVIEW

**Review by EMsnap — APPROVED** [review]

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/KafkaExtractNode.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+    /**
+     * generate table options
+     *
+     * @return options
+     */
+    @Override
+    public Map<String, String> tableOptions() {
+        Map<String, String> options = super.tableOptions();
+        options.put("topic", topic);
```
Comment (healchow) [inline inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/KafkaExtractNode.java:97]: These are magic strings. Can they be extracted as constants?

Comment (gong) [reply inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/KafkaExtractNode.java]: new add kafka option constant

**Review by yunqingmoswu — APPROVED** [review]

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/constant/KafkaConstant.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.apache.inlong.sort.protocol.constant;
+
+/**
+ * kafka option constant
+ */
+public interface KafkaConstant {
```
Comment (EMsnap) [inline inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/constant/KafkaConstant.java:24]: Why interface here

Comment (healchow) [inline inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/constant/KafkaConstant.java:24]: It is suggested to use `class` for Constant, not `interface`.

Comment (gong) [reply inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/constant/KafkaConstant.java:24]: only need constant。 interface is enough

Comment (gong) [reply inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/constant/KafkaConstant.java:24]: Interface is enough. I can change it.

**Review by EMsnap — APPROVED** [review]

**Pull request merged**